src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs

```diff
@@ -1227,6 +1228,201 @@ public static void TaskIDFromExternalContextTest()
             Assert.False(withinTaskId == null || withinTaskId < 0, "Task.CurrentId called from within a Task must be non-negative");
         }
 
+        private static void VerifyCancelTestState(
+            bool gotCancellationException,
+            bool reportedAsCompleted,
+            int actuallyCompletedCount)
+        {
+            Assert.Equal(reportedAsCompleted, !gotCancellationException);
+            if (reportedAsCompleted)
+            {
+                Assert.Equal(500, actuallyCompletedCount);
```

Comment:
It takes some scanning to figure out where this 500 came from.  It'd be better if it could be passed through from the test input into this verifier as the expected count.